app/src/ui/lib/augmented-filter-list.tsx

```diff
@@ -277,11 +279,25 @@ export class AugmentedSectionFilterList<
     }
 
     if (this.props.onFilterListResultsChanged !== undefined) {
-      const itemCount = this.state.rows
+      const oldfilteredItems = prevState.rows
+        .flat()
+        .filter(row => row.kind === 'item')
+        .map(r => r.item)
+
+      const currentFilteredItemIds = this.state.rows
         .flat()
-        .filter(row => row.kind === 'item').length
+        .filter(row => row.kind === 'item')
+        .map(r => r.item)
 
-      this.props.onFilterListResultsChanged(itemCount)
+      if (
+        oldfilteredItems.length !== currentFilteredItemIds.length ||
+        xor(
```

Comment:
Another request for an explanatory comment here. I have a hard time wrapping my head around what this is doing